modules/c++/six.sicd/include/six/sicd/Utilities.h

```diff
@@ -19,29 +19,140 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <utility>
 
 #include <scene/SceneGeometry.h>
 #include <scene/ProjectionModel.h>
-#include <six/sicd/ComplexData.h>
-
-namespace six
-{
-namespace sicd
-{
-class Utilities
-{
-public:
+#include <six/sicd/ComplexData.h>
+
+#include <six/NITFReadControl.h>
+
+namespace six
+{
+namespace sicd
+{
+class Utilities
+{
+public:
     static scene::SceneGeometry* getSceneGeometry(const ComplexData* data);
-
-    static scene::ProjectionModel* getProjectionModel(const ComplexData* data, 
+
+    static scene::ProjectionModel* getProjectionModel(const ComplexData* data, 
             const scene::SceneGeometry* geom);
 
+   
+    /*
+     * getLoadedNitfReadControl creates and loads a NITFReadControl for the
+     * specified file.
+     *
+     * \param sicdPathname SICD NITF pathname
+     * \param schemaPaths One or more files or directories containing SICD
+     * schemas
+     *
+     * \return a NITFReadControl 'load()'ed with the given SICD
+     *
+     */
+    static
+    NITFReadControl getLoadedNITFReadControl(const std::string& sicdPathname,
+                                 const std::vector<std::string>& schemaPaths);
+
+    /*
+     * SicdContents is a helper object for readSicdContents
+     *
+     * This class is a simple wrapper around dynamically allocated
+     * SICD information designed to allow access to either or both 
+     * variables returned by readSicdContents.
+     *
+     */
+    struct SicdContents {
+        // allocated with new
+        ComplexData * complexData;
+        // allocated with new[]
+        std::complex<float>* widebandData;
+    };
+
+    /*
+     * Given a SICD path name and a list of schema, this function reads
+     * and parses the SICD in order to provide the wideband data as well
+     * as the ComplexData associated with the image..
+     *
+     * \param sicdPathname SICD NITF pathname
+     * \param schemaPaths One or more files or directories containing SICD 
+     * schemas
+     *
+     * \return ComplexData associated with the SICD NITF,
+     *           allocated with new
+     *         std::complex<float> buffer containing SICD wideband data,
+     *           allocated with new[]
+     *
+     * \throws See six::sicd::Utilities::getComplexData and 
+     *           six::sicd::Utilities::getWidebandData
+     *
+     */
+    static SicdContents readSicd(const std::string& sicdPathname,
+                                 const std::vector<std::string>& schemaPaths);
+
+    /*
+     * Given a reference to a loaded NITFReadControl, this function
+     * parses the SICD's XML and returns a new ComplexData object.
+     *
+     * \param reader A NITFReadControl loaded with the desired SICD
+     *
+     * \return ComplexData new ComplexData associated with the SICD NITF
+     *
+     * \throws except::Exception if the provided reader is not a SICD
+     *
+     */
+    static ComplexData* getComplexData(NITFReadControl& reader);
```

Comment:
Is there a reason to use auto_ptr over something from mem in coda-oss? I avoided auto_ptr purposefully because I didn't want to add more code that we know is deprecated in C++11.